### ISSUE TITLE
Kernel metadata - replace evaluator_shape with gh_shape

### ISSUE THREAD

**Issue by arporter** [stfc/PSyclone#45]
Following changes in the LFRic infrastructure around evaluators and quadrature, there is a requirement to re-name evaluator_shape to gh_shape.

**Comment by arporter** [comment]
Doing this work on branch `rename_evaluator`.

**Comment by rupertford** [comment]
I accidentally merged the rename_evaluator branch to master.
I then reverted it via the GitHub option.
After that I'm not quite sure what happened - I think I tried to merge master back to rename_evaluator after updating master and ended up with all changed from rename_evaluator removed.
Even if I go back to an earlier version of rename_evaluator I am not able to merge it again to master as it removes any of the rename_evaluator changes.
Apparently the standard solution is to revert the revert (on master I presume) and then merge in any subsequent changes. However, I'm not sure about doing this given that I've already subsequently merged something else to master.
Therefore I've taken a manual approach. I've created a new branch rename_evaluator_2 from the latest version of master and have then copied over the modified files from the original rename_evaluator branch. I will then create a new pull request.
